core/src/main/java/com/linecorp/armeria/client/HttpClientPipelineConfigurator.java

```diff
@@ -231,7 +231,7 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws E
                 if (handshakeFailed &&
                     cause instanceof DecoderException &&
                     cause.getCause() instanceof SSLException) {
-                    // Swallow an SSLException raised after handshake failure.
+                    ctx.channel().attr(PENDING_EXCEPTION).set(cause);
```

Comment:
`.set(cause.getCause()` because we are interested in `SSLException`, not `DecoderException`.